storm-client/src/jvm/org/apache/storm/daemon/worker/BackPressureTracker.java

```diff
@@ -39,6 +40,7 @@
 public class BackPressureTracker {
     static final Logger LOG = LoggerFactory.getLogger(BackPressureTracker.class);
     private final Map<Integer, BackpressureState> tasks;
+    private final Map<Integer, Integer> lastOverflowCount;
```

Comment:
I think we could optimize a bit here by having a single map of taskId -> BackPressureState, and just add the last overflow count to the BackPressureState object. That way, transferLocalBatch can look up the BackPressureState once rather than looking up in both maps here.

Reply:
I agree that it would be cleaner to store it in the BackpressureState object, but can you clarify what you mean by transferLocalBatch will only need to look it up once?  The BackpressureState object isn't exposed beyond the BackPressureTracker class, so you are paying for two map lookups either way. Unless i'm misunderstanding?

Reply:
Yes, you are right, but I think we could make BackPressureStatus public, and change the BackPressureTracker API so we have a `getBackPressureStatus` method for getting the status, and then `recordBackPressure` and `setLastOverflowCount` and similar methods could just take the `BackPressureStatus` instance. Alternatively we could put `recordBackPressure` and similar methods in `BackPressureStatus`.

If we do either of these things, we can get away with only looking up the BackPressureStatus once.